analysis/exponential.lean

```diff
@@ -127,6 +127,22 @@ by simp [log, lt_irrefl]
 @[simp] lemma log_one : log 1 = 0 :=
 exp_injective $ by rw [exp_log zero_lt_one, exp_zero]
 
+noncomputable definition nth_root (x : ℝ) (n : ℕ) : ℝ := exp ((log x) / n)
+
+theorem exp_mul (x : ℝ) : ∀ n : ℕ, exp(n*x) = (exp(x))^n
+| 0 := by rw [nat.cast_zero, zero_mul, exp_zero, pow_zero]
+| (nat.succ n) := by rw [pow_succ', nat.cast_add_one, add_mul, exp_add, ←exp_mul, one_mul]
+
+theorem nth_root_pos {x n} : nth_root x n > 0 := exp_pos _
+
+theorem nth_root_power {x : ℝ} {n} (Hxpos : 0 < x) (Hnpos : 0 < n): (nth_root x n) ^ n = x := 
+by rw [nth_root, ←exp_mul, mul_div_cancel', exp_log Hxpos]; rw nat.cast_ne_zero; exact ne_of_gt Hnpos
+
+theorem nth_root_unique (x y : ℝ) (n : ℕ)
+  (Hxpos : 0 < x) (Hypos : 0 < y) (Hnpos : 0 < n) (Hynx : y ^ n = x) :
+  y = nth_root x n :=
+pow_eq Hypos nth_root_pos Hnpos (Hynx.trans (nth_root_power Hxpos Hnpos).symm)
+

```

Comment:
rather than having `y ^ n = x` as a hypothesis, just prove `nth_root (y ^. n) n = y`. Also `x`, `y` and `n` should be implicit.

Reply:
@ChrisHughes24 Ok, I've made the changes.